### DISCUSSION TITLE
Unable to use css tokens, media queries or any stitches feature

### DISCUSSION THREAD

**Discussion by krishna-sharma-dev** [heroui-inc/heroui#675] [Help] [answered]
Hello,
First things first, I love the NextUI and the way it is built. It seems like one of the top-contenders for other css-in-js frameworks and component library. Having gone through the documentation, i convinced my colleagues at the new company to migrate our recently built website to NextUI. They agreed for it and i am supposed to build a prototype for them by this week.

**The Issue:**
Even though i have followed the steps in the documentation, i am unable to get stitches token or media queries work in the `css={{}}` property.
Here is the code:
<img width="708" alt="Screenshot 2022-08-18 at 12 36 13 PM" src="https://user-images.githubusercontent.com/110056286/185328398-3e535f03-0257-41b2-ba91-8a9b85667cd3.png">

And here is how it shows up on the browser:
<img width="327" alt="Screenshot 2022-08-18 at 12 30 13 PM" src="https://user-images.githubusercontent.com/110056286/185328285-091324ea-dfa2-4af3-98a0-566300f6dc4a.png">

Here is how my _app.tsx and look like:
<img width="701" alt="Screenshot 2022-08-18 at 1 10 47 PM" src="https://user-images.githubusercontent.com/110056286/185338221-012f4cc6-72c7-474d-a734-8952a2a3df52.png">

HEre is how my _document.tsx look like:
<img width="690" alt="Screenshot 2022-08-18 at 1 11 34 PM" src="https://user-images.githubusercontent.com/110056286/185338280-fd17852f-0812-4a37-bf04-48355bd146e5.png">

Neither the color props nor the media queries get applied.

I need help since i might be missing something. Please help in figuring this one out 😢

**Comment by jrgarciadev** [comment, marked as answer]
Hi 👋🏻 @krishna-sharma-dev , I'm glad you liked NextUI, could you please share the React/Next version you are using?

**Reply by krishna-sharma-dev** [reply]
Hi @jrgarciadev , Thanks for the quick reply.
Here are the package versions that i am currently using: 
`"@nextui-org/react@^1.0.0-beta.9"`
`next-12.2.3`
`react-dom@^18.2.0`
`react@^18.2.0`

**Reply by krishna-sharma-dev** [reply]
Also, @jrgarciadev , i tried downgrading to 
- NextUI - 1.0.0-beta.8
- React 17.0.2
- react-dom 17.0.2
But this did not solve the issue. Still getting the same error as this: https://user-images.githubusercontent.com/110056286/185328285-091324ea-dfa2-4af3-98a0-566300f6dc4a.png

**Reply by krishna-sharma-dev** [reply]
So, i guess i figured out the issue @jrgarciadev .
I am using styled-components along with NextUI for my next.js project

I have this code in my next.config.js
`const nextConfig = {
   compiler: {
    styledComponents: true,
  },
  }
  `
  Enabling styled-components this way actually makes my css stitches useless. I tried the same with the create-next-app example in the NextUI's repo and could reproduce the same.
  
 **With styledComponents: true in the example app**
<img width="1440" alt="Screenshot 2022-08-19 at 10 30 40 AM" src="https://user-images.githubusercontent.com/110056286/185546242-c16758b8-85bf-4be4-8f45-db03c014e45d.png">

**With styledComponents removed from the next.config.js.compiler**
<img width="1428" alt="Screenshot 2022-08-19 at 10 31 56 AM" src="https://user-images.githubusercontent.com/110056286/185546369-047395fd-f01b-4682-97ee-85a777d51317.png">
  
  Now my question is, how do i get both styled-components and css stitches working together?

**Reply by krishna-sharma-dev** [reply]
Finally, got it working. Instead of adding `styledComponents:true` in the next.config.js file, i edit my _document.tsx after following [this](https://github.com/vercel/next.js/blob/deprecated-main/examples/with-styled-components/pages/_document.js) guide like this:

```import React from 'react';
import Document, { Html, Head, Main, NextScript } from 'next/document';
import { CssBaseline } from '@nextui-org/react';
import { ServerStyleSheet } from 'styled-components';

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            sheet.collectStyles(<App {...props} />),
        });
      const initialProps = await Document.getInitialProps(ctx);
      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            {sheet.getStyleElement()}
          </>
        ),
      };
    } finally {
      sheet.seal();
    }
  }

  render() {
    return (
      <Html lang="en">
        <Head>{CssBaseline.flush()}</Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```
BTW, i still couldn't figure out why enabling styledComponents through next.config.js was causing this issue.

But nonetheless, i hope this helps others who might face similar issue.
@jrgarciadev

**Reply by jrgarciadev** [reply]
Thanks a lot for sharing the solution @krishna-sharma-dev 🙌🏼
